girder-dandi-archive/girder_dandi_archive/__init__.py

```diff
@@ -28,3 +30,14 @@ def load(self, info):
         Setting().set(UserQuotaPluginSettings.DEFAULT_USER_QUOTA, 0)
 
         info["apiRoot"].dandi = DandiResource()
+
+        events.bind("model.folder.save", "folder_save_listener", locking.folder_save_listener)
+        # TODO item save
+        # TODO file save
+        events.bind(
+            "model.upload.assetstore",
+            "upload_assetstore_listener",
+            locking.upload_assetstore_listener,
+        )
+        # TODO file delete
+        # TODO file move
```

Comment:
Are these TODOs still being worked on in this PR? And if not or if you plan on adding them in a later PR, we should make an issue for them.

Reply:
The ticket was for the bare minimum required for a locking solution, so I deliberately didn't implement these things yet. As I understand it this is a prototype that we don't necessarily want to invest too much time and effort into, so as long as it works well enough we don't need to guarantee perfect operation in all conditions.